src/Swarm/TUI/Controller.hs

```diff
@@ -454,56 +454,81 @@ getNormalizedCurrentScenarioPath =
       gs <- use $ gameState . scenarios
       Just <$> liftIO (normalizeScenarioPath gs p')
 
+saveScenarioInfoOnFinish :: (MonadIO m, MonadState AppState m) => FilePath -> m (Maybe ScenarioInfo)
+saveScenarioInfoOnFinish p = do
```

Comment:
This is just a cut-and-paste of the body of `saveScenarioInfoOnFinishNocheat` (formerly `saveScenarioInfoOnFinish`).